uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlement.php

```diff
@@ -199,12 +199,13 @@ private static function buildLeavePeriodParamsFromCalculation(
     $absenceTypeID = $calculation->getAbsenceType()->id;
     $contactID = $calculation->getContact()['id'];
     $absencePeriodID = $calculation->getAbsencePeriod()->id;
+    $entitlementOverridden = isset($overriddenEntitlement) && trim($overriddenEntitlement) != '';
 
     $params = [
       'type_id' => $absenceTypeID,
       'contact_id' => $contactID,
       'period_id' => $absencePeriodID,
-      'overridden' => (boolean)$overriddenEntitlement,
+      'overridden' => $entitlementOverridden,
```

Comment:
The overridden field is a boolean, that's why we had the cast. Unfortunately, when civi creates the schema, it treats booleans as TinyInt(4). This means that, if you pass an integer here, like 5 for example, that is what will be saved in the database. Please add the cast back.

Reply:
`$entitlementOverridden = isset($overriddenEntitlement) && trim($overriddenEntitlement) != '';` returns a boolean already